proof-systems/src/darlin/poly-commit/src/domain_extended/mod.rs

```diff
@@ -190,4 +232,50 @@ impl<G: Group, PC: PolynomialCommitment<G, Commitment = G>> PolynomialCommitment
     fn challenge_to_scalar(chal: Vec<bool>) -> Result<G::ScalarField, Self::Error> {
         PC::challenge_to_scalar(chal)
     }
+
+    #[cfg(feature = "circuit-friendly")]
+    fn single_point_multi_poly_open<'a>(
+        ck: &Self::CommitterKey,
+        labeled_polynomials: impl IntoIterator<Item=&'a LabeledPolynomial<G::ScalarField>>,
+        point: G::ScalarField,
+        fs_rng: &mut Self::RandomOracle,
+        labeled_randomnesses: impl IntoIterator<Item=&'a LabeledRandomness<Self::Randomness>>,
+        rng: Option<&mut dyn RngCore>
+    ) -> Result<Self::Proof, Self::Error> {
+        /*
+        Iterate over labeled polynomials and values, sorting them in ascending order
+        on the number of segments. The label of the polynomial is employed as a sorting criteria
+        for the polynomials with the same number of segments
+        */
+
+        // employ a counter to check that all polynomials/values are placed in `sorted_collections`,
+        // as in case there are duplicates the `collect` on `BTreeMap`
+        // just stops processing elements of the iterator rather than returning an error
+        let mut counter = 0;
+        let sorted_collections = labeled_polynomials.into_iter().zip(labeled_randomnesses.into_iter()).map(|(poly, rand)| {
+            counter += 1;
+            ((compute_num_of_segments::<G, PC>(ck, poly), poly.label()) , (poly, rand))
+        }
+        ).collect::<BTreeMap<(_, _), (_,_)>>();
+
+        assert_eq!(counter, sorted_collections.len());
+
+        let (sorted_polys, sorted_rands): (Vec<_>, Vec<_>) = sorted_collections.iter().rev().map(|(_, (poly, rand))| (poly, rand)).unzip();
```

Comment:
Minor thing: Perhaps you can re-use the sort_commitments_and_values macro if you generalize it a bit (it can also be a function and not a macro actually)

Reply:
I generalized the macro (and also rename it) to be usable also in the multi-point multi-poly open (i.e., with polynomials and randomnesses). Not sure that it is convenient to have a function instead of a macro since there is no common trait that can be employed for the input operands, we would need to add to the inputs of the function some closures (with probably several generic parameters) for the functions that are needed (i.e., getting a label for a polynomial/commitment)

Reply:
Ok. As an exercise might be worth trying to make this a function and not a macro (passing a function to a macro is a bit weird but I guess it's ok :D)

Reply:
Ok, I was able to replace both macros employed for sorting with functions